include/aspect/simulator/assemblers/interface.h

```diff
@@ -110,6 +110,7 @@ namespace aspect
           std::vector<types::global_dof_index> local_dof_indices;
           std::vector<unsigned int>            dof_component_indices;
           std::vector<SymmetricTensor<2,dim> > grads_phi_u;
+          std::vector<Tensor<2,dim> >          grad_phi_u;
```

Comment:
Sounds good, you know best which formulation makes more sense. I guess the two things to consider are
(1) Which one is more reasonable physically? (Have you discussed that with Cedric? Would it make sense to write a comment about that somewhere?)
(2) What do other people use? (The way I understand your comment above, right at the start of the pull request discussion, other people use the real gradient instead of the symmetric gradient, so one question would be if we need the real gradient to reproduce benchmarks. And I guess we've learned it the hard way with the BA/ALA/TALA approximations that users don't trust the code unless there's the option to reproduce exactly what other people have done.)